File: src/Domain/Entities/Accountant/UpcomingExpense.cs
using System;

namespace Domain.Entities.Accountant;

public class UpcomingExpense : Entity
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; }
    public string Description { get; set; }
    public DateTime Date { get; set; }
    public bool Generated { get; set; }
}

File: src/Clients/accountant2/src/lib/models/entities/upcomingExpense.ts
import type { Syncable } from '$lib/models/syncable';

export class UpcomingExpense implements Syncable {
	categoryName: string | undefined;
	synced = false;

	constructor(
		public id: number,
		public categoryId: number | null,
		public amount: number,
		public currency: string,
		public description: string | null,
		public date: string,
		public generated: boolean,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}
